src/logic/safe/transactions/txHistory.ts

```diff
@@ -2,13 +2,19 @@ import { GnosisSafe } from 'src/types/contracts/gnosis_safe.d'
 import { _getChainId } from 'src/config'
 
 import { checksumAddress } from 'src/utils/checksumAddress'
-import { proposeTransaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import { MultisigTransactionRequest, proposeTransaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
 import { GATEWAY_URL } from 'src/utils/constants'
+import { TxArgs } from '../store/models/types/transaction'
 
-const calculateBodyFrom = async (
-  safeInstance: GnosisSafe,
+type ProposeTxBodyProps = Omit<MultisigTransactionRequest, 'safeTxHash'> & {
+  safeInstance: GnosisSafe
+  data: string | number[]
+}
+
+const calculateBodyFrom = async ({
+  safeInstance,
   to,
-  valueInWei,
+  value,
```

Comment:
I wouldn't change the argument name. It is clearer as `valueInWei`.

Reply:
In this case I tried to be coherent with the types that come from `type MultisigTransactionRequest` . Plus, it is not an exported function so it doesn't change much.

Reply:
This also affects the `getTransactionHash()` method though.

Reply:
But here we are calling the contract method directly, not the `getTransactionHash()`

Reply:
I  just wanted the methods to be typed . Can leave the names as before it it causes confusion

Reply:
On line 31, you pass the `value` to the Safe instance.

Reply:
You can leave it as is, but I would personally refrain from changes like this in the future. I just feel like it removes semantic clarity.